app/controllers/stories/feeds_controller.rb

```diff
@@ -34,22 +34,43 @@ def assign_feed_stories
     end
 
     def signed_in_base_feed
-      if Settings::UserExperience.feed_strategy == "basic"
-        Articles::Feeds::Basic.new(user: current_user, page: @page, tag: params[:tag])
-          .more_comments_minimal_weight_randomized
-      else
-        Articles::Feeds::LargeForemExperimental.new(user: current_user, page: @page, tag: params[:tag])
-          .more_comments_minimal_weight_randomized
+      strategy = AbExperiment.get(experiment: :feed_strategy, controller: self, user: current_user,
+                                  default_value: "original")
+      feed = if strategy.weighted_query_strategy?
+               Articles::Feeds::WeightedQueryStrategy.new(user: current_user, page: @page, tags: params[:tag])
+             elsif Settings::UserExperience.feed_strategy == "basic"
+               Articles::Feeds::Basic.new(user: current_user, page: @page, tag: params[:tag])
+             else
+               Articles::Feeds::LargeForemExperimental.new(user: current_user, page: @page, tag: params[:tag])
+             end
+      Datadog.tracer.trace("#{self.class}.#{__method__}", span_type: "feed", resource: feed.class.to_s) do
```

Comment:
Thank you for adding tracing here! 🙌🏼

I was going to offer some feedback the field values here but on a second reading I think what you've got here is a lot closer to what I was going to suggest than I first thought.

The only thing I think won't be ingested by Datadog the way you intend is the `span_type` (specifically, I think that field is supposed to be an enum of `http`, `db`, `cache`, or `custom`), but I'm not even sure on that and I'm willing to see how it goes. 😄

Reply:
I'd definitely say it's a `db` related thing.  And can adjust accordingly.